pkg/tbtcpg/moving_funds.go

```diff
@@ -0,0 +1,629 @@
+	// start block at `0`.
+	filterStartBlock := uint64(0)
+	if currentBlockNumber > MovingFundsCommitmentLookBackBlocks {
+		filterStartBlock = currentBlockNumber - MovingFundsCommitmentLookBackBlocks
+	}
+
+	filter := &tbtc.MovingFundsCommitmentSubmittedEventFilter{
+		StartBlock:          filterStartBlock,
+		WalletPublicKeyHash: [][20]byte{sourceWalletPublicKeyHash},
+	}
+
+	events, err := mft.chain.PastMovingFundsCommitmentSubmittedEvents(filter)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to get past moving funds commitment submitted events: [%w]",
+			err,
+		)
+	}
+
+	// Moving funds commitment can be submitted only once for a given wallet.
+	// Check just in case.
+	if len(events) != 1 {
+		return nil, fmt.Errorf(
+			"unexpected number of moving funds commitment submitted events: [%v]",
+			len(events),
+		)
+	}
+
+	targetWallets := events[0].TargetWallets
+
+	// Just in case check if the hash of the target wallets matches the moving
+	// funds target wallets commitment hash.
+	calculatedHash := mft.chain.ComputeMovingFundsCommitmentHash(targetWallets)
+	if calculatedHash != targetWalletsCommitmentHash {
+		return nil, ErrWrongCommitmentHash
+	}
+
+	return targetWallets, nil
+}
+
+// GetWalletMembersInfo returns the wallet member IDs based on the provided
+// wallet operator addresses. Additionally, it returns the position of the
+// moving funds task execution operator on the list.
+func (mft *MovingFundsTask) GetWalletMembersInfo(
+	walletOperators []chain.Address,
+	executingOperator chain.Address,
+) ([]uint32, uint32, error) {
+	// Cache mapping operator addresses to their wallet member IDs. It helps to
+	// limit the number of calls to the ETH client if some operator addresses
+	// occur on the list multiple times.
+	operatorIDCache := make(map[chain.Address]uint32)
+	// TODO: Consider adding a global cache at the `ProposalGenerator` level.
+
+	walletMemberIndex := 0
+	walletMemberIDs := make([]uint32, 0)
+
+	for index, operatorAddress := range walletOperators {
+		// If the operator address is the address of the executing operator save
+		// its position. Note that since the executing operator can control
+		// multiple wallet members its address can occur on the list multiple
+		// times. For clarity, we should save the first occurrence on the list,
+		// i.e. when `walletMemberIndex` still holds the value of `0`.
+		if operatorAddress == executingOperator && walletMemberIndex == 0 {
+			// Increment the index by 1 as operator indexing starts at 1, not 0.
+			// This ensures the operator's position is correctly identified in
+			// the range [1, walletOperators.length].
+			walletMemberIndex = index + 1
+		}
+
+		// Search for the operator address in the cache. Store the operator
+		// address in the cache if it's not there.
+		if operatorID, found := operatorIDCache[operatorAddress]; !found {
+			operatorID, err := mft.chain.GetOperatorID(operatorAddress)
```

Comment:
Non-blocking and we can address this in the next PR: Let's rename this `operatorID` to `fetchedOperatorID` to not shadow the variable declared in the `if`.